app/src/main/java/com/delirium/playlistmaker/search/ui/activity/SearchFragment.kt

```diff
@@ -75,19 +87,19 @@ class SearchActivity : AppCompatActivity(), ClickListener {
             }
         }
 
-        viewModel.getHistoryLiveData().observe(this) { history ->
+        viewModel.getHistoryLiveData().observe(viewLifecycleOwner) { history ->
             renderHistory(history.asList())
         }
 
-        binding.recyclerSongs.layoutManager = LinearLayoutManager(this)
+        binding.recyclerSongs.layoutManager = LinearLayoutManager(requireContext())
         binding.recyclerSongs.adapter = adapter
 
         binding.clearSearch.setOnClickListener { it ->
             binding.editSearch.text.clear()
             it.visibility = View.INVISIBLE
 
-            this.currentFocus?.let {
-                val keyboard = getSystemService(Context.INPUT_METHOD_SERVICE) as InputMethodManager
+            requireActivity().currentFocus?.let {
+                val keyboard = requireActivity().getSystemService(Context.INPUT_METHOD_SERVICE) as InputMethodManager
```

Comment:
🍏 
Здесь можно использовать ближайший доступный контекст - контекст фрагмента:

```suggestion
                val keyboard = requireContext().getSystemService(Context.INPUT_METHOD_SERVICE) as InputMethodManager
```